app/battle_preparation/battle_preparation.py

```diff
@@ -0,0 +1,17 @@
+from app.knights.knight import Knight
+
+
+def knights_power_calculation(knight: Knight) -> Knight:

```

Comment:
You could make this function as a method for the `Knight` class since it operates on the `Knight` class instance. Place it in the `Knight` class and rename it to `apply_equipment` to be more descriptive since `power` is not the only entity worked upon.